mobile/src/home-view.js

```diff
@@ -60,68 +62,89 @@ export default class HomeView extends Component {
-        : <View style={{flex: 1}}><WebView ref={input => this.webview = input} style={s.web} source={{uri: "https://react-barrating-widget-cxgvnp.stackblitz.io/"}} injectedJavaScript={injectedJavaScript} onMessage={e => this.saveResults(e.nativeEvent.data)} onLoad={this.sendInfo}/><Button onPress={()=>this.setState({showTable: true, config: "", configKey: ""})} title=""/></View> 
+        : <View style={s.container}><WebView ref={input => this.webview = input} onError={error=>console.log} style={s.web} source={{html: surveyViewHtml}}  injectedJavaScript={this.injectedJavaScript()} onMessage={e => this.saveResults(e.nativeEvent.data)} onLoad={this.sendInfo}/><TouchableOpacity style={s.backButton} onPress={()=>this.setState({showTable: true, config: "", configKey: ""})}/></View> 
         }
       </KeyboardAvoidingView>
     )
   }
 
+
   sendInfo = () => {
-    this.webview.postMessage(this.state.config)
+    const config = JSON.stringify({survey: this.state.config, color: this.state.primaryColor})
+    this.webview.postMessage(config)
   }
 
   closeSurveyModal = () => {
-    console.log(this.state.config)
     this.setState({showTable: false})
   }
 
   saveResults = (resultsString) => {
-    const results = JSON.parse(resultsString)
-    fbc.database.private.adminableUserRef('results').child(this.state.configKey).push({
-      results, creator: client.currentUser, timeTaken: new Date().getTime()
+    let results = JSON.parse(resultsString) 
+    results = Object.values(results)
+    let newQuestionsArray = []
+    let config = JSON.parse(this.state.config)
+    config.pages.forEach(page => {
+      newQuestionsArray = newQuestionsArray.concat(page.elements)
+    })
+    let newResults = []
+    results.forEach((item, index) => {
+      newResults.push({question : newQuestionsArray[index].title, answer: item})
+    })
+    this.props.fbc.database.private.adminableUserRef('results').child(this.state.configKey).push({
+      newResults, creator: this.state.currentUser, timeTaken: new Date().getTime()
       })
     .then(() => this.setState({}))
     .catch (x => console.error(x))    
   }
 
   selectSurvey = (item) => {
-    this.setState({config: item.info, configKey: item.key, disable: false})
+    let parsedInfo = JSON.parse(item.info)
+    parsedInfo.pages.forEach(page => {
+      page.elements.forEach(question => {
+        if (question.choices) {
+          question.choices.forEach(item => {
+            if (item.value) {
+              item.value = item.text
+            }
+          })
+        }
+      })
+    })
+    this.setState({config: JSON.stringify(parsedInfo), configKey: item.key, disable: false})
   }
 
-
+  injectedJavaScript = () => `
+  window.config = {survey: null, color: '${this.state.primaryColor}'}
+  window.document.addEventListener("message", function(e) {
```

Comment:
How does this interact with mobile-web/src/App.js ?

Reply:
I'm not entirely sure but the surveys did not load when I removed. Perhaps we can discuss this further/